src/api/schema/components/mod.rs

```diff
@@ -203,9 +204,11 @@ impl ComponentsQuery {
         .await
     }
 
-    /// Gets a configured component by name
-    async fn component_by_name(&self, name: String) -> Option<Component> {
-        component_by_name(&name)
+    /// Gets a configured component by id
+    async fn component_by_id(&self, name: String) -> Option<Component> {
```

Comment:
for backward compatibility, we might need to consider having 2 functions:
- `component_by_name` which would _only_ give global components
- `component_by_id` which would allow to select by `ComponentId`.

Reply:
Just noting the likely merge conflicts with https://github.com/timberio/vector/pull/8417 (merged 13 hours ago) which refactored the component name field to `component_by_component_id`.

Any API changes also need reflecting in the API clients, including `tap` and `top`. #8417 shows a good example of where code was changed to accommodate the client features.

To ensure the API and its client remain in sync, run `make build-graphql-schema` after making API changes and then attempt to `cargo build` again. The compiler should alert you to anything that's fallen out of sync.